### DISCUSSION TITLE
Start/Stop Enclosure Fan How-To/Help?

### DISCUSSION THREAD

**Discussion by bluong77** [tagyoureit/nodejs-poolController#938] [Q&A] [answered]
Hi,

I am new to owning a swimming pool and have read quite a lot on this wiki (NJS, REM, DASH, etc..) and have started to build my Stand-alone-Nixie Pool Controller. I know nothing about Node Red or programming or Home Automation aside from buying off the shelf things but pretty good at following instructions.

**I have the following pool equipment;**
Pool size: 13,000 Gallon (Single Body)
Max-E-Therm 400k Heater -> Fireman Switch -> Relay Hat
Intellichlor IC20 -> RS485 (BAS Hat)
Argon Lights (Can operate in Intellibrite mode) -> Relay Hat
SuperMax VS Pump (2023 model) -> RS485 (BAS Hat)

**Nixie Stand-Alone Equipment**
RPI4
Sequent 8 Relay Hat
Sequent Building Automation Hat
24V/DC - Power RPI4 and Hat
Buck Converter -> Use to convert 24V to 12V
Enclosure Box with 12V vent Fan
2x 10K Temp Sensor -> BAS Hat

"Ignore the messing wiring in the image.  Will clean it up once i am done the testing phase"

![IMG_2531 Large](https://github.com/tagyoureit/nodejs-poolController/assets/168031944/71b23921-ce28-4c04-9963-191ee3e15765)

I follow most of the Stand-alone-Nixie guide and think i have installed  NJS, Rem, Dashpanel correctly.  As the pool is brand new, I haven't connected the Nixie Pool Controller to any hardware officially yet as i want to run the pool to make sure it is working properly without any automation. I did quickly test to connect the SuperMax VS Pump via RS485 and it works (as Intelliflo VS Pump).

I have tons of questions on setting up the pool within Dashpanel but I wanted to tackle a small task with the Enclosure first.  There is a built-in 12V fan.  I have the 24V/DC powering it via a Buck Converter to drop the voltage to 12V.  This means the fan will always be ON.  I then wired the fan to one of the relay on the sequent hat.  Using REM, i can turn the fan on/off from the sequent 8 Relay Hat tab.  I then added a "Feature" on Dash Panel so there is a button to turn the vent fan on/off.

So to get to the point...
How can i have the vent fan turn on/off based on the Raspberry Pi's CPU temperature or even a temperature sensor attached to the BAS card?  Can it be done with the vent fan connected  to one of the relays on the sequent hat? Is there an option in  Njs pool controller? or does the fan need to be connected to a GPIO pin?  If so, does someone have a script of that.  Ideally, if there is a way within NJS pool controller, that would be great.  Hopefully someone can point me in the right direction.

FYI; I connected the SuperMax VS Pump via RS485 to the Sequent BAS using the Pentair Automation cable (black). I have it set to Intelliflo VS pump in Dashpanel..  I can confirm that the SuperMax VS Pump works and communicates via RS485 so you don't need to go the route of "Digital Inputs / wiring to relays".  Maybe helpful for someone that has this same pump and wondering if RS485 communications works or not.

Thanks
Billy

**Comment by tagyoureit** [comment]
Looking good so far.  You can set up a generic device in REM that will read the CPU temperature.  

![2024-04-29_14-16-11](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/e396331c-c3a6-415f-9280-243824a37624)

Then you set up a feed to one of your relays and it will automatically turn on/off your fan.  No connection to njsPC is needed.

![2024-04-29_14-17-00](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/d35102d8-9280-4954-9e91-bf97d5bc0cad)

**Reply by bluong77** [reply]
Thank you for that info;

I tried to add a Generic device called "CPU Temp" just like the above screenshot.  I configured the feed again just like your example  and even written the code below in your example, assuming the code means if the temperature is > than 90 degree, then turn on the relay?.  When i click "Save",  REM "greys out and Freezes for a bit (30secs)", then when it becomes responsive again, the feed i added is no longer listed.

Is there something that needs to be setup for my Multi-Relay Board? How does it know what relay to turn on in my Sequent Multi-relay hat?

Here are some screen shots to help.

Some screen shots

![Screenshot 2024-04-30 at 10 32 18 AM](https://github.com/tagyoureit/nodejs-poolController/assets/168031944/b71c5394-f8d0-4902-adea-76f47db5c5ed)
![Screenshot 2024-04-30 at 10 32 28 AM](https://github.com/tagyoureit/nodejs-poolController/assets/168031944/fcf511d2-bd61-4899-ad65-f7d0b69b0cfa)
![Screenshot 2024-04-30 at 10 33 18 AM](https://github.com/tagyoureit/nodejs-poolController/assets/168031944/e1369256-5695-4bb8-872c-d38ee28772a5)

**Reply by FlaMike** [reply, marked as answer]
I use relay 5 on my board, not 4.  My feed section is a bit different than yours.  Built my Nixie recently & struggled with this a bit, as well.  You may want to try what worked for me, hopefully it will get you there.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/f71175fa-dd51-4d30-8c11-28313129b41b)

**Reply by bluong77** [reply]
Thanks @FlaMike 
for some reason, @tagyoureit script locks REM and i can't do anything however using your script, it works and the fan is turned on when it hits the assigned temperature. 

Thank you all!

**Reply by FlaMike** [reply]
So glad you got it going!  Enjoy your Nixie--nice looking build :-)

**Reply by bluong77** [reply]
Thanks @FlaMike I am sure to have a bunch more questions once i actually hook up all the wiring to the various pool equipment.

**Reply by FlaMike** [reply]
My pleasure!  Having just gone through the process, I'm *way* too familiar with all of the mistakes one can make along the way (and as a result, I learned a ton of stuff).  For me, it was a big & challenging project.  I'm not fully at the goal yet, but getting a little closer each day.  I'm also in the (early) phase of writing up my experience--part 1 having been posted a couple of days ago.  Gotta get back into writing mode once life permits.  The Nixie fan challenge slipped from my mind until I  saw your post.  Gotta remember to include it...

BTW, folks involved in this project are super helpful and pretty darned  responsive, especially @tagyoureit and @rstrouse.  I'm just a little pawn in game of life.  :-)

**Reply by theRealenabl3D** [reply]
Hello, I've tried both solutions posted here and they are not working.
I have the Sequent 4REL4IN v4.
FYI, the Chem Controller relay feed to NodejsPC works fine.

**Reply by FlaMike** [reply]
Can you post your generic device settings/feeds, etc?  It's difficult to tell what's going on without that information.  Screen shots, per above should help.

**Reply by theRealenabl3D** [reply]
Here are my settings:
![settings](https://github.com/user-attachments/assets/9378f915-7e84-4118-9d38-e585d6ab1c37)

And here's the error I get in REM:
error: Sequent 4rel4in v4 error setting I/O relayState channel e out of range.

Thanks

**Reply by FlaMike** [reply]
Gosh, I'm not sure what to suggest.  I don't have any experience with the relay board you're using.  I assume the fan is connected to relay #1 on the board.  Is relay 1 configured?  Can you turn the fan on/off by clicking on the relay icon?

**Reply by theRealenabl3D** [reply]
Yes, all relays on the board are enabled and respond using REM's GUI, see below:
![fan](https://github.com/user-attachments/assets/1104a09a-836d-4832-b1fd-77956b3695d4)

Below are from the logs:
error: Sequent 4rel4in v4 error setting I/O relayState channel e out of range.
verbose: Feed internal/Internal Devices sending relayState: {"id":1,"state":true} to i2c:1:2
debug: 14 - Sequent 4rel4in v4 Executed read command 0x03 byte read:0x00
debug: 14 - Sequent 4rel4in v4 Executed read command 0x00 byte read:0x00
verbose: Emitting: /chemController : {"acidPump":false,"id":50}
verbose: Feed njspc/PoolController sending acidPump: false to chemController

**Reply by FlaMike** [reply]
Unfortunately, I'm already at/beyond my knowledge in this area.  The only other thing I can think of at the moment is the relay's configuration.  Mine is below--I'm hoping there is some minor difference from your configuration that just might be material.  If not, perhaps someone with greater knowledge will chime in & help you out.  

Just curious...  What temperature is your CPU reporting?  Is it reasonable?

Good  luck!  I'll keep an eye on the thread & hopefully learn what ultimately works for you.

![image](https://github.com/user-attachments/assets/2e6cf694-f007-46fe-b23d-6b237449041f)

**Reply by theRealenabl3D** [reply]
Board configuration looks the same. CPU reports 115-120degF.

**Reply by rstrouse** [reply]
Please pull REM.  I do not have a 4rel4in board but another user has one and have provided a pull request to fix a couple of issues with the relays.

**Reply by theRealenabl3D** [reply]
Thanks for all the help... but unfortunately I'm still getting the same error after a "git pull"...

**Reply by rstrouse** [reply]
Just to be sure are you able to turn the relay on and off with the user interface.

**Reply by theRealenabl3D** [reply]
Yes. I can command the relays with both the REM GUI and Sequent CLI.
FYI, this is the version reported using CLI:
```
rel4in -v
4rel4in v1.0.2 Copyright (c) 2016 - 2022 Sequent Microsystems
```
Versus the one shown in REM:
![version](https://github.com/user-attachments/assets/50f3052e-c467-4d55-88b8-03c9d8beb3b4)

Link to the board I got: https://sequentmicrosystems.com/collections/industrial-automation/products/four-relays-four-inputs-for-raspberry-pi

Another FYI, list of CLI commands:
```
4rel4in -h
	-v		Display the 4rel4in command version number
	-h		Display the list of command options or one command option details
	-warranty	Display the warranty
	-list:		List all 4rel4in boards connected
			return the # of boards and stack level for every board
	reltest:		Turn ON and OFF the relays until press a key
	relrd:		Read relays  status
	relwr:		Set relays (AC switch) On/Off
	inrd:		Read inputs  status
	acinrd:		Read inputs status when the signal is AC voltage
	blrd:		Read the button latch, return 1 if the button has been pushed since last read
	brd:		Read the button current state, 1 = pushed, 0 = released
	cfgcntrd:		Read the configuration of the count function
	cfgcntwr:		Set counting function On/Off
	cntrd:		Read the counter of one input channel
	cntrst:		Reset the counter of one input channel
	cntppsrd:		Read the counter pulse per second of one input channel
	cfgencrd:		Read the configuration of the quadrature encoder function
	cfgencwr:		Set quadrature encoder function On/Off
	encrd:		Read the counter of one encoder channel
	encrst:		Reset the counter of one encoder input channel
	pwmrd:		Read the pwm fill factor(%) for one input channel
	freqrd:		Read the frequency(Hz) for one input channel
	cfg485rd:	Read the RS485 communication settings
	cfg485wr:	Write the RS485 communication settings
	cfgextird:		Read external interupt enable status
	cfgextiwr:		Set external interrupt generation On/Off
```

**Comment by FlaMike** [comment]
Wow!  I am truly stumped.  I don't know if folks other than me have not
responded to you because I marked the issue as answered a while ago?.
Perhaps others may chime in if you open a new thread & reference this one.
If I think of anything else, I will certainly post my thoughts, but at the
moment, I'm at a dead end.  Sorry!
